src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -614,13 +619,14 @@ private void VerifyPrecompileStatus(IEnumerable<FunctionDescriptor> functions)
             }
         }
 
-        private static void AddDirectTypes(List<Type> types, Collection<FunctionDescriptor> functions)
+        // Get the set of types that should be directly loaded. These have the "configurationSource" : "attributes" set.
+        // They will be indexed and invoked directly by the WebJobs SDK and skip the IL generator and invoker paths.
+        private static IEnumerable<Type> GetDirectTypes(IEnumerable<FunctionMetadata> functionMetadatas)
         {
             HashSet<Type> visitedTypes = new HashSet<Type>();
 
-            foreach (var function in functions)
+            foreach (var metadata in functionMetadatas)
```

Comment:
@watashiSHUN had a similar change proposed. Can we have a different name, other than `functionMetadatas`? (Something that doesn't use "metadatas")